### DISCUSSION TITLE
czg存在缓存，eslint错误还是错误，实际上已经解决？

### DISCUSSION THREAD

**Discussion by ht-sauce** [Zhengqbbb/cz-git#65] [Q&A]
1、我需要先使用git commit刷新缓存，否则每次czg都是错误的。对应的eslint校验都是上一次的错误信息。但是我独立运行eslint --fix等是没有错误信息的。
当手动git commit之后，再次运行czg则正常提交成功
2、有什么办法直接git commit触发校验吗？比如我只要提交的格式符合规范即可

**Comment by Zhengqbbb** [comment]
> 1、我需要先使用git commit刷新缓存，否则每次czg都是错误的。对应的eslint校验都是上一次的错误信息。但是我独立运行eslint --fix等是没有错误信息的。 当手动git commit之后，再次运行czg则正常提交成功 2、有什么办法直接git commit触发校验吗？比如我只要提交的格式符合规范即可

是执行完 `czg` CLI 后 eslint 校验失败吗？
这个很奇怪能否知道你的 hook 或 lint-staged 是怎么写的，能贴一下吗，正常来说既然到校验了说明 czg 已经退出了

> 2、有什么办法直接git commit触发校验吗？比如我只要提交的格式符合规范即可

https://blog.qbb.sh/post/2022/01/11/husky/#git-hooks

是 `pre-commit` git hook 吗？
https://github.com/Zhengqbbb/cz-git/blob/3fc9d67bbf3adf5150f3961a44c01d8445799cc8/package.json#L52-L54

**Reply by ht-sauce** [reply]
![image](https://user-images.githubusercontent.com/34828964/191151150-f000bd0b-d466-450f-b5f3-2baaf1489274.png)
我是这样处理的。请问有什么问题吗？

**Reply by Zhengqbbb** [reply]
husky 的 `pre-commit` 配置定义是 `pnpm lint-staged` 即可

**Reply by ht-sauce** [reply]
![image](https://user-images.githubusercontent.com/34828964/191152290-83d95d74-445a-4dd2-90f7-adb97d0bf2e5.png)
![image](https://user-images.githubusercontent.com/34828964/191152304-61c80f11-1cdd-4745-94e6-2860d7df7691.png)
我这里是这样定义的。似乎和package.json并没有差异

**Reply by ht-sauce** [reply]
同时该问题也会出现在普通的czg上，当我变更一个文件的时候。我直接运行czg命令，会提示我当前项目没有发生变更
![image](https://user-images.githubusercontent.com/34828964/191153467-8daabad6-2607-48cc-bf70-dfc072d24f9a.png)
![image](https://user-images.githubusercontent.com/34828964/191153483-b153cde8-965f-497a-8b62-ba3d26411743.png)
我感觉是我配置错了。但是又不知道是哪里引起的问题。

**Reply by Zhengqbbb** [reply]
这个是你没有`git add` 命令 暂存区检查失败出来的
试试
```sh
git add .
czg
```
or
```sh
# 仅修改文件没有新增文件，自动 add 进行缓存区
czg -a
```

**Reply by Zhengqbbb** [reply]
嗯嗯
我比较好奇为什么eslint的命令要使用
` --max-warnings 0`
--max-warnings Int              Number of warnings to trigger nonzero exit code - default: -1
有什么特殊的使用吗

如果可以的话试试看
```json
{
  "lint-staged": {
    "*.{vue,js,ts,jsx,cjs,mjs,json,tsx,md,cts,mts}": "eslint --fix --ignore-path .gitignore"
  },
}
```

```diff
- pnpm exec
+ pnpm
```

实测有效，如果不行的话贴图我看看
但是大概率是eslint 命令写法的问题了

<img width="986" alt="image" src="https://user-images.githubusercontent.com/40693636/191154594-f7dcc234-1be9-425c-9aeb-0b75f5066c99.png">

**Reply by ht-sauce** [reply]
这个其实是我抄element-plus的。但是实际上只是测试一下。并不是导致问题的原因。
实际上是您刚才给出的方案，需要在执行命令中加入git add等。刷新git提交的缓存。否则直接run czg依旧是检查上一个改动的项目情况。
不知道element plus团队如何解决的。我也是照猫画虎。谢谢，我理解原因所在了

**Reply by Zhengqbbb** [reply]
> 这个其实是我抄element-plus的。但是实际上只是测试一下。并不是导致问题的原因。 实际上是您刚才给出的方案，需要在执行命令中加入git add等。刷新git提交的缓存。否则直接run czg依旧是检查上一个改动的项目情况。 不知道element plus团队如何解决的。我也是照猫画虎。谢谢，我理解原因所在了

ok  需要我提供PR吗  我看见你的项目了
git 的 commit commit不是变更文件而是 暂存区的所有文件，看来的 蓝色高亮 的不够明显 2333 )

**Reply by ht-sauce** [reply]
我想您可以尝试修复这个问题。不过我理解原理了，可以改配置文件提前运行命令来解决。感谢解答

**Reply by Zhengqbbb** [reply]
> 我想您可以尝试修复这个问题。不过我理解原理了，可以改配置文件提前运行命令来解决。感谢解答

👌，不过这不是bug
- `git add`  `git commit` 是两个命令
- 我提供的 git commit 的解决方案
- 用户是要 git add 全部  git add 部分 git add 单个文件是用户决定的
